imports/ui/components/welcome-dialog.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Dialog from 'material-ui/Dialog'
+import FontIcon from 'material-ui/FontIcon'
+import TextField from 'material-ui/TextField'
+import RaisedButton from 'material-ui/RaisedButton'
+import UneeTIcon from './unee-t-icon'
+
+import { closeDialogButtonStyle } from './generic-dialog.mui-styles'
+import {
+  textInputFloatingLabelStyle,
+  textInputStyle,
+  textInputUnderlineFocusStyle
+} from './form-controls.mui-styles'
+
+class WelcomeDialog extends Component {
+  constructor () {
+    super(...arguments)
+    this.state = {
+      firstName: '',
+      lastName: ''
+    }
+  }
+  render () {
+    const {
+      show,
+      onDismissed,
+      onNameSubmitted,
+      invitedByDetails,
+      caseItem: { id: caseId, summary: caseSummary },
+      unitItem: { name: unitName }
+    } = this.props
+    const {
+      firstName,
+      lastName
+    } = this.state
+    return (
+      <Dialog
+        open={show}
+        modal
+      >
+        <button onClick={onDismissed}
+          className='button b--none bg-transparent absolute top-1 pt2 right-1 outline-0'
+        >
+          <FontIcon className='material-icons' style={closeDialogButtonStyle}>close</FontIcon>
+        </button>
+        <div className='tc'>
+          <UneeTIcon isDarkType style={{width: 67, height: 67}} />
+          <div className='mt2 pt1 bondi-blue f3 fw5'>
+            Welcome to Unee-T!
+          </div>
+        </div>
+        <div className='lh-copy mt2 pt1'>
+          {invitedByDetails ? invitedByDetails.name : ''} has invited you to collaborate on the case&nbsp;
+          <span className='b'>
+            "#{caseId} - {caseSummary}"
+          </span>
+          &nbsp;in&nbsp;
+          <span className='b'>
+            {unitName}
+          </span>
+          <br />
+          <br />
```

Comment:
code between line 63 and 76 could be replaced with something like:
+          <br />
+          Please enter your name
+        </div>
+        <TextField
+          floatingLabelText='First name'
+          fullWidth
+          floatingLabelShrinkStyle={textInputFloatingLabelStyle}
+          inputStyle={textInputStyle}
+          underlineFocusStyle={textInputUnderlineFocusStyle}
+          onChange={e => this.setState({
+            firstName: e.target.value
+          })}
+          value={firstName}
+        />
+        <TextField
+          floatingLabelText='Last name'
+          fullWidth
+          floatingLabelShrinkStyle={textInputFloatingLabelStyle}
+          inputStyle={textInputStyle}
+          underlineFocusStyle={textInputUnderlineFocusStyle}
+          onChange={e => this.setState({
+            lastName: e.target.value
+          })}
+          value={lastName}
+        />
+		<br>
+		Why we need this:
+		</br>
+		<br>
+		Unee-T hides your email address from other users. If you do not enter your first and last name, we will use the beginnign of your email address as your alias.
+		</br>
+		<br>
+		This can be confusing and it's better to use an alias that other user will be able to recognize.
+		</br>